### DISCUSSION TITLE
Add a Delay send

### DISCUSSION THREAD

**Discussion by Qui-Gon-JooL** [SynthstromAudible/DelugeFirmware#1472] [Ideas]
Hi,

  It would really be welcome to modify the delay by adding a "delay send" function to be able to cut off the delay sends without cutting the volume of the delay itself, or an alternative delay which would allow this.

  In general on groove boxes the delay is a send FX, this allows for example to send small bursts of delay on a track without modifying the volume of the delay, here on the deluge, when I open the delay for a note is taken from the delay, it works, but as soon as I cut the delay, the sound of the delay also cuts off and does not continue in the background.

  Another example would be for the build up at the end of the beat, we turn up the delay, everything is fine but when we want to cut the sending of the delay to balance the climax, the delay cuts off, it should play a little more in the background. ..

**Comment by manysounds** [comment]
YES please. Dub style delays!

**Comment by Qui-Gon-JooL** [comment]
This seems possible and no one ever looks into it because it's too complicated ? 
Or is this material impossible to put in place ?

**Reply by sapphire-arches** [reply]
It's possible to implement, just maybe slightly more challenging than it initially appears.

 - The voice code is currently extremely fragile and complex. We need better testing so we can refactor it to a more reasonable state.
 - It's not clear what the UI for this should be. The existing delay effect is per-clip, so how would a user distinguish between sending to the global delay buffer and the per-clip buffer?

**Comment by Qui-Gon-JooL** [comment]
Thank you for the answer.

I don't know what a voice code is? I have never programmed
But I trust you, it won't be easy.

Do you mean that you have already worked to implement it? if possible after testing or you say that it would be good to think about perhaps one day starting by doing more in-depth tests... ? :) I mean, I wonder if it could happen in the next little while or if it's unlikely to happen within the year ?

I don't know if it can help but I would like to see a delay master menu in the dive menu where we adjust all the parameters like time\feedback\volume, and for example in the choice of digital or analog delay there could be a third option "master" which would send to a master delay track.

Which would be a master track like the reverb.
And on each track a send which would replace the current amount button.

**Reply by sapphire-arches** [reply]
I have worked on some of the required infrastructure, yes. I can't give a timeline, it depends on whether someone is motivated to work on the feature (we're all volunteers with varying availability).

Exposing the master delay as a delay type is a good idea for the UI!

**Reply by Qui-Gon-JooL** [reply]
Yes I understand, and thank you for your immense work you have already made the Deluge a war machine :)

**Comment by DW-James** [comment]
Yes please to Delay (and ideally Reverb) working like sends rather than inserts!
It wouldn't necessarily need to be made as a global effect that all tracks can actually send to, but could be done a change to the architecture of the existing track effect so it acted in that way. 
At the moment it seems delay 'Amount' is essentially a control for Delay Feedback. 
There has to be both Wet/Dry routes set up at the moment, otherwise when you pressed a note you wouldn't hear the initial hit.
I think what's needed is simply a gain control adding into the input of the Delay's Wet route, but outside of the feedback loop so that it doesn't effect the repeats. That way Delay Amount would control that gain and become 'How much of our signal is going through the delay' and the Feedback would become a separate and distinct setting for 'How long the delays repeat for'. It's a bit clumsy currently and prone to getting overloaded and mushy as presumably wet/dry is constantly set to 50/50 and all you can control is the feedback. 
The challenge I think is how to integrate this into what's currently there on the interface and whether it's possible to do without breaking existing songs. 
There was a spare Delay pad, but that's become Sync (which does seem a little weird to have both SYNC and RATE when they're sort of the same thing) Otherwise I would love to see AMOUNT become the SEND TO DELAY gain and the spare pad repurposed to be a shortcut for accessing the Feedback.

**Comment by Dust-r** [comment]
I would really like this feature of Delay being a send effect instead of an insert. I'd like to do a proper dub mix on the Deluge.

**Reply by Dust-r** [reply]
I don't really need the delay time adjustment feature on the top Gold Knob; changing the delay time live rarely works, one wants to set it and forget it. Instead, the bottom Gold Knob could be the delay send, and the top Gold Knob the feedback control/volume.

**Comment by betonizm** [comment]
+1. And it would be nice to make the ping pong delay adjustable. Not just Disabled/Enabled, but with a bipolar parameter, where 0 is Disable, and to the right or left it adjusts the direction and value - width. How this is done in elektron groovebox (parameter width)

**Comment by alielquartje** [comment]
I'd love control over Delay send amount too if possible. I did a deluge live performance last year and really missed delay send for live fx. I find it very difficult to control with only the intensity/feedback control.

**Comment by worker-gray** [comment]
PLEASE! Give us pre-delay, number of repeats (one being the minimum) and wet/dry mix too. A nice slapback or flam on the drum would be most SPICY.
